### ISSUE TITLE
SocialSharing popup only shows when I close and open the app

### ISSUE THREAD

**Issue by dshaw002** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#703]
I have the current version of SocialSharing (5.1.3) with Ionic 2 and latest Cordova.

On my phone iOS 10, when I open the app and tap the button to trigger the event, nothing happens. If I leave the app and open it again, however, the share popup shows.

Any ideas what's causing this? Other events don't have this issue.

**Comment by Quadrophone** [comment]
I was able to fix this by adding `gap://ready` to the default-src section of the Content Security Policy

**Comment by shoebrizwanathar** [comment]
Yeah! It worked for me as well!! 

I am working on Ionic and Cordova project and was struggling with this issue.
By adding gap://ready to the default-src section of the Content Security Policy works for opening share sheet smoothly without any blockage or hiding behind and also it works for Email Composer to open smoothly.

I have tested it on iOS 10.1 version iPhone and android.

Use this in your index.html in head section:

<meta http-equiv="Content-Security-Policy" content="default-src * gap://ready; style-src 'self' 'unsafe-inline'; script-src * 'unsafe-inline' 'unsafe-eval'">

**Comment by leolandotan** [comment]
This seems to be an issue still on Ionic v1 and iOS 10.3.

**Comment by mateusbello** [comment]
I have the exact same problem but the the solutions above didn't work.

I have

<meta http-equiv="Content-Security-Policy" content="default-src * gap://ready data: 'unsafe-eval'; img-src * data: gap://ready; frame-src *; connect-src * ; style-src * 'self' 'unsafe-inline' 'unsafe-eval'; media-src *; script-src * data: 'self' 'unsafe-inline' 'unsafe-eval' gap://ready; ">
  
and

 <allow-intent href="http://*/*" />
  <allow-intent href="https://*/*" />
  <allow-intent href="tel:*" />
  <allow-intent href="sms:*" />
  <allow-intent href="mailto:*" />
  <allow-intent href="geo:*" />

**Comment by vinicoliveira** [comment]
Hello. How did you solve it? I am still getting error!!!!

**Comment by mateusbello** [comment]
For me it was a matter of adding

   ` <meta http-equiv="Content-Security-Policy" content="default-src * gap://ready data: 'unsafe-eval'; img-src * data: gap://ready; frame-src *; connect-src * ; style-src * 'self' 'unsafe-inline' 'unsafe-eval'; media-src *; script-src * data: 'self' 'unsafe-inline' 'unsafe-eval' gap://ready; ">`

And changing the config.xml to have

```
  <allow-navigation href="*://*.mysite.com/*" />
  <allow-intent href="*" />
  <allow-navigation href="*" />
```

**Comment by vinicoliveira** [comment]
Hello. Thanks for your fast answer =]
But it is still not working.

No errors...it just does not launch the sheet with apps.

` $cordovaSocialSharing
          .share("My MSG", My Title, '', 'https://mylink')
          .then(function(result) {
            alert(result);
          }, function(err) {
           alert(err);
          });`

No  alerts.

I put the meta tag in index.html
And I addded the <allow> in config.xml

What is wrong?

**Comment by mateusbello** [comment]
If you close and open the app, does it work ? If not, it is something else not related to the issue on this topic.

**Comment by vinicoliveira** [comment]
no. it does not work. By the way, none of the $cordovaSocialSharing are working on iOS. I cannot get any error.

**Comment by vinicoliveira** [comment]
Let me update my answer..

The thing is... if I press the share button nothing happens. But if I press the share button and then focus an input the sheet appears.

**Comment by mateusbello** [comment]
Sorry but this is not related to this topic. Could you please create another one ? Thanks ... I'm not sure what is the issue you are having btw, but I guess it is not related to security.
